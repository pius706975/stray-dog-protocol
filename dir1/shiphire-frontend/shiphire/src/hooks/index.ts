export * from './useAuth';
export * from './useRenter';
export * from './useShip';
export * from './useUser';
export * from './useHelper';
export * from './useShipOwner';
export * from './useAdmin';
export * from './useMutationWithErrorHandler';
