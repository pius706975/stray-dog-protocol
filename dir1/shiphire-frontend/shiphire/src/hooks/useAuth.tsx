import { AxiosError, AxiosResponse } from 'axios';
import useMutationWithErrorHandler from './useMutationWithErrorHandler';
import { authService } from '../services';
import {
    AppleSignInRequest,
    DefaultResponse,
    GetNewTokenResponse,
    GoogleSignInRequest,
    ResetPasswordRequest,
    SignInRequest,
    SignInResponse,
    SignOutRequest,
    SignUpRequest,
    VerifyForgotPassOTPRequest,
} from '../types';

export const useSignUp = () =>
    useMutationWithErrorHandler<
        AxiosResponse<DefaultResponse>,
        AxiosError,
        SignUpRequest
    >(data => authService.signUp(data));

export const useSignIn = () =>
    useMutationWithErrorHandler<
        AxiosResponse<SignInResponse>,
        AxiosError,
        SignInRequest
    >(data => authService.signIn(data));

export const useGetNewToken = () =>
    useMutationWithErrorHandler<
        AxiosResponse<GetNewTokenResponse>,
        AxiosError,
        void
    >(() => authService.getNewToken());

export const useSignOut = () =>
    useMutationWithErrorHandler<
        AxiosResponse<DefaultResponse>,
        AxiosError,
        SignOutRequest
    >(data => authService.signOut(data));

export const useSendOTPForgotPass = () =>
    useMutationWithErrorHandler<
        AxiosResponse<DefaultResponse>,
        AxiosError,
        { email: string }
    >(data => authService.sendOTPForgotPass(data));

export const useVerifyForgotPassOTP = () =>
    useMutationWithErrorHandler<
        AxiosResponse<DefaultResponse>,
        AxiosError,
        VerifyForgotPassOTPRequest
    >(data => authService.verifyForgotPassOTP(data));

export const useResetPassword = () =>
    useMutationWithErrorHandler<
        AxiosResponse<DefaultResponse>,
        AxiosError,
        ResetPasswordRequest
    >(data => authService.resetPassword(data));

export const useGoogleSignIn = () =>
    useMutationWithErrorHandler<
        AxiosResponse<SignInResponse>,
        AxiosError,
        GoogleSignInRequest
    >(data => authService.googleSignIn(data));

export const useAppleSignIn = () =>
    useMutationWithErrorHandler<
        AxiosResponse<SignInResponse>,
        AxiosError,
        AppleSignInRequest
    >(data => authService.appleSignIn(data));
