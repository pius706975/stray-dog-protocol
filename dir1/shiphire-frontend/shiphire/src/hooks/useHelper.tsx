import { AxiosError, AxiosResponse } from 'axios';
import useMutationWithErrorHandler from './useMutationWithErrorHandler';
import { helperService } from '../services';
import {
    DefaultResponse,
    GetShipFacilityRequest,
    GetShipFacilityResponse,
    GetShipLocationsResponse,
    GetShipSpesificationRequest,
    GetShipSpesificationResponse,
    GetUserLocationRequest,
    GetUserLocationResponse,
} from '../types';

export const useGetShipFacility = () =>
    useMutationWithErrorHandler<
        AxiosResponse<GetShipFacilityResponse>,
        AxiosError,
        GetShipFacilityRequest
    >(data => {
        return helperService.getShipFacilityByShipType(data);
    });

export const useGetShipSpesification = () =>
    useMutationWithErrorHandler<
        AxiosResponse<GetShipSpesificationResponse>,
        AxiosError,
        GetShipSpesificationRequest
    >(data => {
        return helperService.getShipSpesificationByShipType(data);
    });

export const useGetShipLocations = () =>
    useMutationWithErrorHandler<
        AxiosResponse<GetShipLocationsResponse>,
        AxiosError,
        void
    >(helperService.getShipLocations);

export const useGetUserLocation = () =>
    useMutationWithErrorHandler<
        AxiosResponse<GetUserLocationResponse>,
        AxiosError,
        GetUserLocationRequest
    >(data => {
        return helperService.getUserLocation(data);
    });

export const useGetHealthCheck = () =>
    useMutationWithErrorHandler<
        AxiosResponse<DefaultResponse>,
        AxiosError,
        void
    >(helperService.getHealthCheck);
