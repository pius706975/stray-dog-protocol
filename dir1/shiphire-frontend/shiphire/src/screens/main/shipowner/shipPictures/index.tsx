export { default as ShipPictures } from './ShipPictures';
