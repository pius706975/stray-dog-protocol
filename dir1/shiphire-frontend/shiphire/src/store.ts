import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
