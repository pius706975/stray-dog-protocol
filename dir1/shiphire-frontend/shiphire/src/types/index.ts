export * from './Components'
export * from './Navigators'
export * from './ReduxToolkit'
export * from './Screens'
export * from './Services'