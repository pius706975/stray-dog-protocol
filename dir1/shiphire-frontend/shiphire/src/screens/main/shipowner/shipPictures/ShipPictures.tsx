import React from 'react';
import { View, Text, TouchableOpacity, Image } from 'react-native-ui-lib';
import { ShipPictureProps } from '../../../../types';
import {
    Button,
    CustomText,
    ScreenLayout,
    TextInput,
} from '../../../../components';
import { FontFamily, FontSize, PlusIcon } from '../../../../configs';
import ImagePicker, { ImageOrVideo } from 'react-native-image-crop-picker';
import mime from 'mime';
import { useDispatch } from 'react-redux';
import { useSubmitShipPicturesBeforeSailing } from '../../../../hooks';
import { modalSlice, progressIndicatorSlice } from '../../../../slices';
import { handleAxiosError } from '../../../../utils';
import { ConfirmationModal } from './components';
import { useTranslation } from 'react-i18next';

const ShipPictures: React.FC<ShipPictureProps> = ({ navigation, route }) => {
    const {
        transactionId,
        sailingStatus,
        beforeSailingPictures,
        afterSailingPictures,
    } = route.params;
    const [selectedImages, setSelectedImages] = React.useState<ImageOrVideo[]>(
        [],
    );
    const [descriptions, setDescriptions] = React.useState<string[]>([]);
    const dispatch = useDispatch();
    const { hideModal, showModal } = modalSlice.actions;
    const { hideProgressIndicator, showProgressIndicator } =
        progressIndicatorSlice.actions;
    const mutationSubmitShipPicturesBeforeSailing =
        useSubmitShipPicturesBeforeSailing();
    const [isConfirmationModalVisible, setConfirmationModalVisible] =
        React.useState(false);
    const { t } = useTranslation('saveship');

    const openImage = (indexToUpdate: number | null = null) => {
        ImagePicker.openPicker({
            multiple: true,
            maxFiles: 10,
            width: 300,
            height: 400,
            cropping: true,
        })
            .then((images: ImageOrVideo[]) => {
                console.log('Images:', images);
                if (
                    indexToUpdate !== null &&
                    indexToUpdate < selectedImages.length
                ) {
                    const updatedImages = [...selectedImages];
                    updatedImages[indexToUpdate] = images[0]; // Only update the selected index
                    setSelectedImages(updatedImages);
                    const updatedDescriptions = [...descriptions];
                    updatedDescriptions[indexToUpdate] = ''; // Reset description if needed
                    setDescriptions(updatedDescriptions);
                } else {
                    console.log('Selected Images:', images);
                    setSelectedImages(images);
                    setDescriptions(Array(images.length).fill(''));
                    console.log('Descriptions:', descriptions);
                }
            })
            .catch(e => console.log(e));
    };

    const submitImages = async () => {
        console.log('Selected Images:', selectedImages);
        console.log('Descriptions:', descriptions);
        const formData = new FormData();
        dispatch(showProgressIndicator());
        formData.append('transactionId', transactionId);
        selectedImages.forEach((image, index) => {
            formData.append('files', {
                uri: image.path,
                type: mime.getType(image.path),
                name: `Image Before ${descriptions[index]}`,
            });

            formData.append('descriptions', descriptions[index]);
        });

        mutationSubmitShipPicturesBeforeSailing.mutate(formData, {
            onSuccess: () => {
                dispatch(hideProgressIndicator());
                dispatch(
                    showModal({
                        status: 'success',
                        text: t('ShipPictures.textSuccessUpload'),
                    }),
                );

                setTimeout(() => {
                    dispatch(hideModal());
                    navigation.navigate('OwnerTransactionTabNav');
                }, 2000);
            },
            onError: err => {
                dispatch(hideProgressIndicator());
                handleAxiosError(err);
                dispatch(
                    showModal({
                        status: 'failed',
                        text: t('ShipPictures.textFailedUpload'),
                    }),
                );

                setTimeout(() => {
                    dispatch(hideModal());
                }, 2000);
            },
        });
    };

    const isFormValid = () => {
        // Check if at least one image is selected
        if (selectedImages.length === 0) {
            return false;
        }

        // Check if all description fields have values
        for (const description of descriptions) {
            if (!description.trim()) {
                return false;
            }
        }

        // If both conditions are met, the form is considered valid
        return true;
    };

    const showConfirmationModal = () => {
        setConfirmationModalVisible(true);
    };

    const hideConfirmationModal = () => {
        setConfirmationModalVisible(false);
    };

    const confirmSubmit = () => {
        hideConfirmationModal();
        submitImages();
    };

    const renderSubmittedImages = () => {
        if (
            beforeSailingPictures &&
            Array.isArray(beforeSailingPictures) &&
            beforeSailingPictures.length > 0
        ) {
            return beforeSailingPictures.map((image, index) => (
                <View key={index} style={{ alignSelf: 'center' }}>
                    <Image
                        source={{ uri: image.documentUrl }}
                        style={{
                            width: 150,
                            height: 150,
                            borderRadius: 100 / 2,
                            marginBottom: 10,
                        }}
                    />
                    <Text style={{ textAlign: 'center', marginBottom: 20 }}>
                        {image.description
                            ? image.description
                            : 'No description'}
                    </Text>
                </View>
            ));
        }
        return null;
    };
    return (
        <ScreenLayout
            testId="ShipPicturesScreen"
            backgroundColor="light"
            padding={10}
            gap={10}>
            {beforeSailingPictures?.length &&
            beforeSailingPictures.length > 0 ? (
                <View
                    style={{
                        padding: 10,
                    }}>
                    <Text
                        style={{
                            fontFamily: FontFamily.regular,
                            fontSize: FontSize.md,
                        }}>
                        {t('ShipPictures.PreviouslySubmitted')}
                    </Text>
                    <View
                        style={{
                            padding: 10,
                        }}>
                        {renderSubmittedImages()}
                    </View>
                </View>
            ) : null}
            <View
                style={{
                    padding: 10,
                }}>
                <Text
                    style={{
                        fontFamily: FontFamily.regular,
                        fontSize: FontSize.md,
                    }}>
                    {t('ShipPictures.PleaseSubmit')}
                </Text>
            </View>

            <View style={{ flex: 1 }}>
                <View>
                    {selectedImages.length > 0 ? (
                        selectedImages.map((image, index) => (
                            <View testID={`SelectedImage-${index}`} key={index}>
                                <TouchableOpacity
                                    style={{
                                        marginTop: 20,
                                        width: 150,
                                        height: 150,
                                        borderRadius: 100 / 2,
                                        alignSelf: 'center',
                                    }}
                                    onPress={() => openImage(index)}>
                                    <Image
                                        source={{ uri: image.path }}
                                        style={{
                                            width: 150,
                                            height: 150,
                                            borderRadius: 100 / 2,
                                            marginBottom: 10,
                                        }}
                                    />
                                </TouchableOpacity>
                                {/* Input field for descriptions */}
                                <TextInput
                                    // placeholder={t(`ShipPictures.textDescription.${index + 1}`)}
                                    placeholder={t(`ShipPictures.textDescription`)}
                                    value={descriptions[index]}
                                    onChange={text => {
                                        const updatedDescriptions = [
                                            ...descriptions,
                                        ];
                                        updatedDescriptions[index] = text;
                                        setDescriptions(updatedDescriptions);
                                        console.log('Index', index + 1);
                                        console.log(
                                            'Descriptions:',
                                            descriptions,
                                        );
                                    }}
                                />
                            </View>
                        ))
                    ) : (
                        <TouchableOpacity
                            testID="AddImageBtn"
                            style={{
                                marginTop: 20,
                                width: 150,
                                height: 150,
                                borderRadius: 100 / 2,
                                alignSelf: 'center',
                            }}
                            onPress={openImage}>
                            <Image
                                source={require('../../../../../assets/images/Empty.jpg')}
                                style={{
                                    width: 150,
                                    height: 150,
                                    borderRadius: 100 / 2,
                                }}
                            />
                            <View
                                style={{
                                    position: 'absolute',
                                    top: '50%',
                                    left: '50%',
                                    transform: [
                                        { translateX: -15 },
                                        { translateY: -15 },
                                    ],
                                    backgroundColor: 'rgba(255, 255, 255, 0.7)',
                                    width: 30,
                                    height: 30,
                                    borderRadius: 15,
                                    alignItems: 'center',
                                    justifyContent: 'center',
                                }}>
                                <PlusIcon />
                            </View>
                        </TouchableOpacity>
                    )}
                </View>
            </View>

            <Button
                testID="SubmitBtn"
                title={t('ShipPictures.Submit')}
                onSubmit={showConfirmationModal}
                disable={!isFormValid()}
            />
            <ConfirmationModal
                isVisible={isConfirmationModalVisible}
                onConfirm={confirmSubmit}
                onCancel={hideConfirmationModal}
            />
        </ScreenLayout>
    );
};

export default ShipPictures;
