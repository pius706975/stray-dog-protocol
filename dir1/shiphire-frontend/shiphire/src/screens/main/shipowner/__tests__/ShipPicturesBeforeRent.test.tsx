import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MainOwnerStackParamList } from '../../../../types';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import {
    setMockImagePickerPathResolve,
    setMockImagePickerResolve,
} from '../../../../jest/setup';
import MockAdapter from 'axios-mock-adapter';
import httpRequest from '../../../../services/api';
import { ShipPictures } from '../shipPictures';
import ImageCropPicker from 'react-native-image-crop-picker';
import { useSubmitShipPicturesBeforeSailing } from '../../../../hooks';

jest.mock('../../../../hooks');
jest.mock('react-native-image-crop-picker', () => ({
    openPicker: jest.fn(() =>
        Promise.resolve([
            {
                creationDate: '1522437259',
                cropRect: null,
                data: null,
                duration: null,
                exif: null,
                filename: 'IMG_0006.HEIC',
                height: 3024,
                localIdentifier: 'CC95F08C-88C3-4012-9D6D-64A413D254B3/L0/001',
                mime: 'image/heic',
                modificationDate: '1530944178',
                path: '/Users/tabrezakhlaque/Library/Developer/CoreSimulator/Devices/3A21CC65-FCBF-4B7B-BE2C-87E18F17D6DC/data/Containers/Data/Application/C34290DF-2011-480C-9076-2B3CCFA8CD17/tmp/react-native-image-crop-picker/B02EB928-5871-4616-BA8F-05EF93037EB1.jpg',
                size: 2808983,
                sourceURL:
                    'file:///Users/tabrezakhlaque/Library/Developer/CoreSimulator/Devices/3A21CC65-FCBF-4B7B-BE2C-87E18F17D6DC/data/Media/DCIM/100APPLE/IMG_0006.HEIC',
                width: 4032,
            },
        ]),
    ),
    types: {
        image: 'image/jpg',
    },
}));

const queryClient = new QueryClient();
const MainOwner = createNativeStackNavigator<MainOwnerStackParamList>();
const mockAdapter = new MockAdapter(httpRequest);

const mockSubmitImages = () => {
    mockAdapter
        .onPost('/ship-owner/submit-ship-pictures-before-sailing')
        .reply(200, {
            message: 'success',
        });
    (useSubmitShipPicturesBeforeSailing as jest.Mock).mockImplementation(
        () => ({
            mutate: jest.fn().mockImplementation((data, options) => {
                const { onSuccess } = options;
                const mockResponseData = {
                    message: 'success',
                };
                onSuccess(mockResponseData);
            }),
        }),
    );
};

const shipPicturesComponent = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainOwner.Navigator>
                    <MainOwner.Screen
                        name="ShipPictures"
                        component={ShipPictures}
                        initialParams={{
                            transactionId: 'transactionId',
                            sailingStatus: 'beforeSailing',
                            beforeSailingPictures: {}[0],
                            afterSailingPictures: {}[0],
                        }}
                    />
                </MainOwner.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

describe('Testing Ship Pictures Before Rent', () => {
    describe('Snapshot Testing', () => {
        it('should render Ship Pictures Before Rent', async () => {
            const shipPictures = render(shipPicturesComponent);
            expect(shipPictures).toMatchSnapshot();
        });
    });
    describe('Unit Testing', () => {
        it('should handle open image picker', async () => {
            const { getByTestId } = render(shipPicturesComponent);

            const selectImageButton = getByTestId('AddImageBtn');
            fireEvent.press(selectImageButton);
            setMockImagePickerResolve(true);
            await waitFor(() => {
                expect(ImageCropPicker.openPicker).toHaveBeenCalled();
            });
        });
        it('should handle success submit ship pictures before sailing', async () => {
            mockSubmitImages();
            const { getByTestId, getByPlaceholderText } = render(
                shipPicturesComponent,
            );
            const selectImageButton = getByTestId('AddImageBtn');
            fireEvent.press(selectImageButton);
            await waitFor(() => {
                setMockImagePickerPathResolve(true);
                const selectedImage = getByTestId('SelectedImage-0');
                expect(selectedImage).toBeTruthy();
            });
            const descriptionInput = getByPlaceholderText('Description 1');
            fireEvent.changeText(descriptionInput, 'Test Description');
            const submitButton = getByTestId('SubmitBtn');

            fireEvent.press(submitButton);
            const confirmButton = getByTestId('confirmButton');
            fireEvent.press(confirmButton);
            await waitFor(() => {
                const modal = store.getState().modal;
                expect(modal).toEqual({
                    status: 'success',
                    text: 'Ship pictures before sailing submitted successfully',
                    visible: true,
                });
                jest.advanceTimersByTime(2000);
            });
        });
        it('should handle error submit ship pictures before sailing', async () => {
            mockAdapter
                .onPost('/ship-owner/submit-ship-pictures-before-sailing')
                .reply(500, {
                    message: 'error',
                });
            (
                useSubmitShipPicturesBeforeSailing as jest.Mock
            ).mockImplementation(() => ({
                mutate: jest.fn().mockImplementation((data, options) => {
                    const { onError } = options;
                    const mockResponseData = {
                        message: 'error',
                    };
                    onError(mockResponseData);
                }),
            }));
            const { getByTestId, getByPlaceholderText } = render(
                shipPicturesComponent,
            );
            const selectImageButton = getByTestId('AddImageBtn');
            fireEvent.press(selectImageButton);
            await waitFor(() => {
                setMockImagePickerPathResolve(true);
                const selectedImage = getByTestId('SelectedImage-0');
                expect(selectedImage).toBeTruthy();
            });
            const descriptionInput = getByPlaceholderText('Description 1');
            fireEvent.changeText(descriptionInput, 'Test Description');
            const submitButton = getByTestId('SubmitBtn');

            fireEvent.press(submitButton);
            const confirmButton = getByTestId('confirmButton');
            fireEvent.press(confirmButton);
            await waitFor(() => {
                const modal = store.getState().modal;
                expect(modal).toEqual({
                    status: 'failed',
                    text: 'Error submitting ship pictures',
                    visible: true,
                });
                jest.advanceTimersByTime(2000);
            });
        });
    });
});
