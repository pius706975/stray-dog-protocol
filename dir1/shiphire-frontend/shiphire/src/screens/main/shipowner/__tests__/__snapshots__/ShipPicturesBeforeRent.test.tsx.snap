// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Ship Pictures Before Rent Snapshot Testing should render Ship Pictures Before Rent 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="ShipPicturesScreen"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": 10,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": 10,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "padding": 10,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "color": "#20303C",
                          "writingDirection": "ltr",
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "fontFamily": "Kanit-Regular",
                          "fontSize": 18,
                        },
                      ]
                    }
                  >
                    ShipPictures.PleaseSubmit
                  </Text>
                </View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "flex": 1,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignSelf": "center",
                          "borderRadius": 50,
                          "height": 150,
                          "marginTop": 20,
                          "opacity": 1,
                          "width": 150,
                        }
                      }
                      testID="AddImageBtn"
                    >
                      <Image
                        accessibilityRole="image"
                        accessible={false}
                        assetGroup="icons"
                        onError={[Function]}
                        source={
                          {
                            "testUri": "../../../assets/images/Empty.jpg",
                          }
                        }
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            false,
                            {
                              "borderRadius": 50,
                              "height": 150,
                              "width": 150,
                            },
                            false,
                          ]
                        }
                      />
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "alignItems": "center",
                              "backgroundColor": "rgba(255, 255, 255, 0.7)",
                              "borderRadius": 15,
                              "height": 30,
                              "justifyContent": "center",
                              "left": "50%",
                              "position": "absolute",
                              "top": "50%",
                              "transform": [
                                {
                                  "translateX": -15,
                                },
                                {
                                  "translateY": -15,
                                },
                              ],
                              "width": 30,
                            },
                          ]
                        }
                      >
                        <RNSVGSvgView
                          align="xMidYMid"
                          bbHeight="24"
                          bbWidth="24"
                          fill="none"
                          focusable={false}
                          height={24}
                          meetOrSlice={0}
                          minX={0}
                          minY={0}
                          style={
                            [
                              {
                                "backgroundColor": "transparent",
                                "borderWidth": 0,
                              },
                              {
                                "flex": 0,
                                "height": 24,
                                "width": 24,
                              },
                            ]
                          }
                          vbHeight={24}
                          vbWidth={24}
                          width={24}
                          xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g clip-path="url(#clip0_77_28470)">
    <path d="M18 13H13V18C13 18.55 12.55 19 12 19C11.45 19 11 18.55 11 18V13H6C5.45 13 5 12.55 5 12C5 11.45 5.45 11 6 11H11V6C11 5.45 11.45 5 12 5C12.55 5 13 5.45 13 6V11H18C18.55 11 19 11.45 19 12C19 12.55 18.55 13 18 13Z" fill="#323232"/>
    </g>
    <defs>
    <clipPath id="clip0_77_28470">
    <rect width="24" height="24" fill="white"/>
    </clipPath>
    </defs>
    </svg>
    
    
    "
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <RNSVGGroup
                            fill={null}
                            propList={
                              [
                                "fill",
                              ]
                            }
                          >
                            <RNSVGGroup
                              clipPath="clip0_77_28470"
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                            >
                              <RNSVGPath
                                d="M18 13H13V18C13 18.55 12.55 19 12 19C11.45 19 11 18.55 11 18V13H6C5.45 13 5 12.55 5 12C5 11.45 5.45 11 6 11H11V6C11 5.45 11.45 5 12 5C12.55 5 13 5.45 13 6V11H18C18.55 11 19 11.45 19 12C19 12.55 18.55 13 18 13Z"
                                fill={
                                  {
                                    "payload": 4281479730,
                                    "type": 0,
                                  }
                                }
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                              />
                            </RNSVGGroup>
                            <RNSVGDefs>
                              <RNSVGClipPath
                                fill={
                                  {
                                    "payload": 4278190080,
                                    "type": 0,
                                  }
                                }
                                name="clip0_77_28470"
                              >
                                <RNSVGRect
                                  fill={
                                    {
                                      "payload": 4294967295,
                                      "type": 0,
                                    }
                                  }
                                  height="24"
                                  propList={
                                    [
                                      "fill",
                                    ]
                                  }
                                  width="24"
                                  x="0"
                                  y="0"
                                />
                              </RNSVGClipPath>
                            </RNSVGDefs>
                          </RNSVGGroup>
                        </RNSVGSvgView>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": true,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#6499c4",
                      "borderRadius": 6,
                      "elevation": 2,
                      "flexDirection": "row",
                      "gap": 0,
                      "justifyContent": "center",
                      "paddingHorizontal": 10,
                      "paddingVertical": 6,
                    }
                  }
                  testID="SubmitBtn"
                >
                  <Text
                    numberOfLines={0}
                    style={
                      {
                        "borderBottomWidth": 0,
                        "borderColor": undefined,
                        "color": "#f0f6fb",
                        "fontFamily": "Kanit-Medium",
                        "fontSize": 18,
                        "lineHeight": undefined,
                        "textAlign": undefined,
                      }
                    }
                  >
                    ShipPictures.Submit
                  </Text>
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "position": "absolute",
                          "right": 20,
                        },
                      ]
                    }
                  />
                </View>
                <Modal
                  animationType="slide"
                  hardwareAccelerated={false}
                  modifiers={
                    {
                      "alignments": {},
                      "backgroundColor": undefined,
                      "borderRadius": undefined,
                      "color": "transparent",
                      "flexStyle": undefined,
                      "margins": {},
                      "paddings": {},
                      "positionStyle": undefined,
                      "typography": undefined,
                    }
                  }
                  onRequestClose={[Function]}
                  transparent={true}
                  visible={false}
                />
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="ShipPictures"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
